dsgrid/config/representative_period_time_dimension_config.py

```diff
@@ -44,10 +48,93 @@ def check_dataset_time_consistency(self, load_data_df):
             self._format_handler.list_expected_dataset_timestamps(self.model.ranges), load_data_df
         )
 
-    def convert_dataframe(self, df, project_time_dim):
-        # TODO: Create a dataframe with timestamps covering project_time_dim.model.ranges
-        # for all model_years in the df, then join the df to that dataframe on the time columns.
-        # Account for time zone.
+    def build_time_dataframe(self):
+        time_cols = self.get_timestamp_load_data_columns()
+        schema = StructType(
+            [StructField(time_col, IntegerType(), False) for time_col in time_cols]
+        )
+
+        model_time = self.list_expected_dataset_timestamps()
+        df_time = _get_spark_session().createDataFrame(model_time, schema=schema)
+
+        return df_time
+
+    # def build_time_dataframe_with_time_zone(self):
+    #     return self.build_time_dataframe()
+
+    def convert_dataframe(self, df=None, project_time_dim=None, time_zone_mapping=None):
+        # in spark.dayofweek: 1=Sunday, 7=Saturday
+        # in pandas.dt.day_of_week: 0=Monday, 6=Sunday
```

Comment:
Pandas is probably not relevant. This is the dsgrid data format (and Python standard library).